src/tracksdata/array/_test/test_graph_array.py

```diff
@@ -171,3 +187,82 @@ def test_graph_array_view_dtype_inference() -> None:
 
     # Dtype should be updated based on the actual data
     assert array_view.dtype == np.float64
+
+
+@fixture(params=[(10, 100, 100), (10, 100, 100, 100)])
+def multi_node_graph_from_image(request) -> GraphArrayView:
+    """Fixture to create a graph with multiple nodes for testing."""
+    shape = request.param
+    label = np.zeros(shape, dtype=np.uint8)
+    for i in range(shape[0]):
+        label[i, 10:20, 10:20] = i + 1
+    graph = RustWorkXGraph()
+    nodes_operator = RegionPropsNodes(extra_properties=["label"])
+    nodes_operator.add_nodes(graph, labels=label)
+    return GraphArrayView(graph=graph, shape=shape, attr_key="label"), label
+
+
+def test_graph_array_view_equal(multi_node_graph_from_image) -> None:
+    array_view, label = multi_node_graph_from_image
+    assert array_view.shape == label.shape
+    for t in range(array_view.shape[0]):
+        assert np.array_equal(array_view[t], label[t])
+    assert np.array_equal(array_view, label)
+    assert np.array_equal(array_view[:5], label[:5])
+    assert np.array_equal(array_view[[1, 6, 7]], label[[1, 6, 7]])
+    assert np.array_equal(array_view[:, 3, 10:20][3, 2:5], label[:, 3, 10:20][3, 2:5])
+    assert array_view.ndim == label.ndim
+    assert array_view.dtype == np.int64  # fixed
+    assert array_view.shape == label.shape
+    # assert np.array_equal(array_view[0], label[0])
+
+
+def test_graph_array_view_getitem_multi_slices(multi_node_graph_from_image) -> None:
+    """Test __getitem__ with slices."""
+    array_view, label = multi_node_graph_from_image
+
+    for count_slice in range(1, array_view.ndim):
+        # Test with slice(10, 20)
+        window = tuple([5] + [slice(10, 20)] * count_slice)
+        assert np.array_equal(array_view[window], label[window])
+        # Test with slice(10, 20, 2)
+        window = tuple([5] + [slice(10, 20, 2)] * count_slice)
+        assert np.array_equal(array_view[window], label[window])
+        # Test with slice(None, 20)
+        window = tuple([5] + [slice(None, 20)] * count_slice)
+        assert np.array_equal(array_view[window], label[window])
+        # Test with slice(10, None)
+        window = tuple([5] + [slice(10, None)] * count_slice)
+        assert np.array_equal(array_view[window], label[window])
+        # Test with slice(None, None)
+        window = tuple([5] + [slice(None, None)] * count_slice)
+        assert np.array_equal(array_view[window], label[window])
+
+
+possible_combinations = [
+    (slice(3, 20), slice(5, None)),
+    (slice(3, 20), slice(None, 15)),
+    (slice(3, 20), 4),
+    (slice(3, 20), [4, 5]),
+    ([5, 6, 7, 8, 9], slice(1, 3)),
```

Comment:
Does it work with non-sequential slicing as `(1, 7, 10, 5)`?
We don't need to support this, but I would  prefer to raise an error rather than wrong results.